src/SourceBuild/content/repo-projects/source-build-reference-packages.proj

```diff
@@ -25,4 +25,24 @@
   </Target>
 
   <Import Project="Sdk.targets" Sdk="Microsoft.NET.Sdk" />
+
+  <UsingTask AssemblyFile="$(LeakDetectionTasksAssembly)" TaskName="MarkAndCatalogPackages" />
+
+  <Target Name="PoisonSbrp"
+          Condition="'$(EnablePoison)' == 'true'"
+          Inputs="$(MSBuildProjectFullPath)"
+          Outputs="$(RepoCompletedSemaphorePath)PoisonCurrentSbrp.complete"
+          AfterTargets="Build">
+    <ItemGroup>
+      <CurrentSbrp Include="$(SourceBuiltPackagesPath)SourceBuildReferencePackages/**/*.nupkg" />
+    </ItemGroup>
+
+    <Message Importance="High" Text="[$([System.DateTime]::Now.ToString('HH:mm:ss.ff'))] Poisoning current SBRP for leak detection." />
+
+    <MarkAndCatalogPackages PackagesToMark="@(CurrentSbrp)" CatalogOutputFilePath="$(CurrentSbrpPoisonReportDataFile)" MarkerFileName="$(CurrentSbrpPoisonMarkerFile)" />
```

Comment:
One thing I noticed about this task is that it doesn't log any diagnostic information.  If you are having troubles with this task, consider updating it to log the packages and assemblies being marked.  This would be a nice addition diagnostically speaking.  If you do this, I would include this in a separate PR.